recipes/_connect_rundeck_api_client.rb

```diff
@@ -1,6 +1,7 @@
 # rest-client gem is needed for api client library
 # install at compile time so library can be used as soon as needed
-include_recipe 'build-essential'
+build-essential
```

Comment:
This needs to be the resource which is "build_essential" not "build-essential"

Try changing it to

```ruby
build_essential 'install compilation tools for gem install'
```